src/JsonHelper.php

```diff
@@ -42,13 +42,13 @@ public static function writeFile(string $filePath, array $data, bool $formatted
             mkdir($filePathDir, 0777, true);
         }
 
-        $result = file_put_contents(
+        $result = @file_put_contents(
             $filePath,
             self::encode($data, $formatted)
         );
 
         if ($result === false) {
```

Comment:
a nemuzem tu podminu dat celou pryc a zavolat radsi v tom testu setEnvironment() ?
ref: https://github.com/keboola/php-component/pull/48#discussion_r225705287

toto je drbani za uchem pres koleno - nastavime handler, kterej zachyti vyjimku, tu ale preskoci a vzapeti rucne vyhodime uplne stejnou vyjimku

Reply:
Ale v tu chvíli to bude závislé na nějakých vnějších faktorech. Třeba když to použiješ ještě v run.php, tak to nemusí fungovat. Leda by se setEnvironment volalo tady uvnitř. To ale může být někdy taky nečekané. 

Ale uznávám, že je minimálně špatně, že to vyhazuje ErrorException. Měla by to být vlastní výjimka, aby šla odchytnout případě.

Reply:
> Ale v tu chvíli to bude závislé na nějakých vnějších faktorech. 

ano, bude. Ale zase je to realistickej zpusob jak se to pouziva

> Třeba když to použiješ ještě v run.php, tak to nemusí fungovat. 

dokud je to soucati php-component tak by me to netrapilo, protoze v php-component se to tak nepouziva a pouzivat se to tak nema. Pokud bysme to vytahli do uplne samostatne knihovny, tak 100% souhlas.

> Leda by se setEnvironment volalo tady uvnitř. To ale může být někdy taky nečekané.

to ne, to je psycho

> Ale uznávám, že je minimálně špatně, že to vyhazuje ErrorException. Měla by to být vlastní výjimka, aby šla odchytnout případě.

jop, to by mi prilo cistsi

Reply:
Upravil jsem to na `JsonHelperException`. Tím pádem je tam opodstatnění pro ten `@`